contracts/BeaconFactory.sol

```diff
@@ -76,4 +79,26 @@ contract BeaconFactory is IIdentityBeaconFactory {
 
         return address(proxy);
     }
+
+    /**
+     * @dev Creates a new identity with the adddress as the owner and registers it with a handle
+     * @param beacon The beacon address to use logic contract resolution
+     * @param owner The initial owner's address of the new contract
+     * @param handle The handle the new identity proxy under which should be registered
+     *
+     * @dev This MUST emit ProxyCreated with the address of the new proxy contract
+     * @dev This must revert if registration reverts
+     */
+    function createAndRegisterBeaconProxy(
+        address beacon,
+        address owner,
+        string calldata handle
+    ) external override {
+        // Create beacon contract
+        address addr = this.createBeaconProxyWithOwner(beacon, owner);
```

Comment:
This call is more expensive because it is calling an "external" function. We should likely only be calling internal functions.

Reply:
I created a private function and call that now. It saves 2k gas.